src/main/java/lux/fm/bookingservice/service/impl/BookingServiceImpl.java

```diff
@@ -29,9 +31,56 @@ public class BookingServiceImpl implements BookingService {
     private final BookingRepository bookingRepository;
     private final BookingMapper bookingMapper;
     private final AccommodationRepository accommodationRepository;
+    private final UserRepository userRepository;
     private final NotificationService notificationService;
     private final PaymentRepository paymentRepository;
 
+    @Override
+    @Scheduled(cron = "0 0 0 * * ?")
+    @Transactional
+    public void checkExpiredBookings() {
+        List<Booking> expiredBookings = bookingRepository.checkExpiredBookings(
+                LocalDate.now().plusDays(1)
```

Comment:
Create separate component class (ScheduledService or something like this) and move this method to it